app/controllers/protocols_controller.rb

```diff
@@ -658,194 +729,374 @@ def export
+    if iterating_key['protocol_name'].present? &&
+       iterating_key['full_name'] &&
+       iterating_key['link']
+      append =
+        t(
+          'protocols.protocols_io_import.comp_append.sub_protocol.title'
+        ) +
+        sanitize_input(iterating_key['protocol_name']) +
+        t(
+          'protocols.protocols_io_import.comp_append.sub_protocol.author'
+        ) +
+        sanitize_input(iterating_key['full_name']) +
+        t('protocols.protocols_io_import.comp_append.general_link') +
+        sanitize_input(iterating_key['link'])
+      return append
+    end
+    ''
+  end
+
+  def pio_stp_19(iterating_key) # protocols io safety information parser
+    if iterating_key['body'].present? &&
+       iterating_key['link']
+      append =
+        t(
+          'protocols.protocols_io_import.comp_append.safety_infor.title'
+        ) +
+        sanitize_input(iterating_key['body']) +
+        t('protocols.protocols_io_import.comp_append.general_link') +
+        sanitize_input(iterating_key['link'])
+      return append
+    end
+    ''
+  end
+
+  def protocols_io_fill_desc(json_hash)
+    description_array = %w[
+      (before_start warning guidelines manuscript_citation publish_date
+      created_on vendor_name vendor_link keywords tags link)
+    ]
+    description_string = sanitize_input(params['protocol']['description'])
+    description_array.each do |e|
+      if e == 'created_on' && json_hash[e].present?
+        new_e = e.humanize
+        description_string +=
+          new_e.to_s + ':  ' +
+          sanitize_input(params['protocol']['created_at'].to_s) + "\n"
+      elsif e == 'tags' && json_hash[e].any? && json_hash[e] != ''
+        new_e = e.humanize
+        description_string +=
+          new_e.to_s + ': '
+        json_hash[e].each do |tag|
+          description_string +=
+            sanitize_input(tag['tag_name']) + ' , '
+        end
+        description_string += "\n"
+        # Since protocols description field doesnt show html,i just remove it
+        # because its even messier (using Sanitize)
+        # what this does is basically appends "FIELD NAME: "+" FIELD VALUE"
+        # to description for various fields
+      elsif json_hash[e].present?
+        new_e = e.humanize
+        description_string +=
+          new_e.to_s + ':  ' +
+          sanitize_input(json_hash[e].to_s) + "\n"
+      end
+    end
+    description_string
+  end
+
+  def protocols_io_fill_step(original_json, newj)
+    # newj = new json
+    # (simple to map) id 1= step description, id 6= section (title),
+    # id 17= expected result
+    # (complex mapping with nested hashes) id 8 = software package,
+    # id 9 = dataset, id 15 = command, id 18 = attached sub protocol
+    # id 19= safety information ,
+    # id 20= regents (materials, like scinote samples kind of)
+    newj['steps'] = {}
+    # I use these 2 variables below to avoid rubocop errors
```

Comment:
I don't like how you are handling this rubocop errors. Remove these 2 variables.
You could create Hash `newj['steps'] = {}` before you call the method `protocols_io_fill_step(original_json, newj)` and then you would pass in the parameter `newj['steps']`. With these all the lines will get shorter.